src/Cake.Core/IO/NuGet/IFrameworkTargetable.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections.Generic;
+using System.Runtime.Versioning;
+
+namespace Cake.Core.IO.NuGet
+{
+    /// <summary>
+    /// Represents an item that may support specific .Net framework versions
+    /// </summary>
+    public interface IFrameworkTargetable
+    {
+        /// <summary>
+        /// Gets the frameworks supported by this item.
+        /// </summary>
+        /// <value>
+        /// The supported frameworks.
+        /// </value>
+        IEnumerable<FrameworkName> SupportedFrameworks { get; }
```

Comment:
As stated above, this is actually borrowed directly from `NuGet.Core.`
Do you think it should be left as they designed it, or do you prefer to change it?